### PR TITLE
find_quadicon: handle cloud instances and fix for archived/orphaned lookups

### PR DESCRIPTION
**Pull request by vrutkovs**
{{pytest: cfme/tests/cloud_infra_common/test_discovery.py --use-provider 'vsphere55' -v --long-running}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit d9d9dedb0cb2ddd255c8bf267b88101f607b9c86:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/common/vm.py [hunk 1]
```diff
@@ -298,8 +298,13 @@ def find_quadicon(
         # paginator.results_per_page(1000)
         if use_search:
             if not search.has_quick_search_box():
-                sel.force_navigate('vm_templates_provider_branch',
-                                   context={'provider_name': self.provider.name})
+                # We don't use provider-specific page (vm_templates_provider_branch) here
+                # as those don't list archived/orphaned VMs
+                if self.provider.category == "infra":
+                    sel.force_navigate('infra_vm_and_templates')
```
Comment (psav) [inline cfme/common/vm.py]: This should a be a string that is set in the base provider classes in the cloud and infra modules.

**Comment by dajoRH** [comment]
Lint report for commit cc1ee1bb2dbef093f0c2e1ac34b307f821d2f3a0:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/common/vm.py [hunk 2]
```diff
@@ -297,10 +297,15 @@ def find_quadicon(
...
         if use_search:
-            if not search.has_quick_search_box():
-                sel.force_navigate('vm_templates_provider_branch',
-                                   context={'provider_name': self.provider.name})
-            search.normal_search(self.name)
+            try:
+                if not search.has_quick_search_box() and self.provider.instances_page_name:
+                    # We don't use provider-specific page (vm_templates_provider_branch) here
+                    # as those don't list archived/orphaned VMs
+                    sel.force_navigate(self.provider.instances_page_name)
```
Comment (psav) [inline cfme/common/vm.py]: Just thinking - there is a ``do_not_navigate`` - should we take that into account here too?

Comment (vrutkovs) [inline cfme/common/vm.py]: Right, we should unset ```use_search``` in this case

**Comment by dajoRH** [comment]
Lint report for commit a99fcdb8c07627717099aac5de4fede46de716e6:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**